wordpress-mobile/WordPress-iOS
# Remove domain dashboard card feature flag and related code

Fixes #20987

- [x] Remove domain dashboard card feature flag
- [x] Remove domain dashboard card-related code

Additionally:
- [x] Remove free to paid plans feature flag, since feature needs to always be enabled

## To test:

2 cases need to be confirmed:

- On the Jetpack app, choose free WP.com site, confirm **free to paid plans card** appears and the **domain dashboard card** doesn't appear
- On the Jetpack app, choose paid WP.com site, confirm **free to paid plans card** and  **domain dashboard card** doesn't appear


## Regression Notes
1. Potential unintended areas of impact

Showing free to paid plans card in wrong cases

2. What I did to test those areas of impact (or what existing automated tests I relied on)

Manual testing, existing unit tests

3. What automated tests I added (or what prevented me from doing so)

Edited existing unit tests

PR submission checklist:


UI Changes testing checklist:
Does not apply
